docs/howto/operator/gcp/dataflow.rst

```diff
@@ -0,0 +1,120 @@
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Google Cloud Dataflow Operators
+===============================
+
+`Dataflow <https://cloud.google.com/dataflow/>`__ is a managed service for
+executing a wide variety of data processing patterns. These pipelines are created
+using the Apache Beam programming model which allows for both batch and streaming.
+
+Airflow provides operators to start/create Java and Python Dataflow jobs.
+
+.. contents::
+  :depth: 1
+  :local:
+
+Prerequisite Tasks
+^^^^^^^^^^^^^^^^^^
+
+.. include:: _partials/prerequisite_tasks.rst
+
+Create a Java Job
+^^^^^^^^^^^^^^^^^
+
+.. _howto/operator:DataflowCreateJavaJobOperator:
+
+Using a JAR on GCS
+""""""""""""""""""
+
+To start a Java Cloud Dataflow job with the JAR on GCS use
+:class:`~airflow.providers.google.cloud.operators.dataflow.DataflowCreateJavaJobOperator`.
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataflow.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_operator_start_java_job]
+    :end-before: [END howto_operator_start_java_job]
+
+Using a JAR on local
+""""""""""""""""""""
+
+To start a Java Cloud Dataflow job with a JAR on local use
+:class:`~airflow.providers.google.cloud.operators.dataflow.DataflowCreateJavaJobOperator`.
+If the JAR is on GCS, you can download it to your local with
+:class:`~airflow.providers.google.cloud.operators.gcs.GCSToLocalOperator`.
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataflow.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_operator_start_local_java_job]
+    :end-before: [END howto_operator_start_local_java_job]
+
+Create a Python Job
+^^^^^^^^^^^^^^^^^^^
+
+.. _howto/operator:DataflowCreatePythonJobOperator:
+
+Using a Python file on GCS
+""""""""""""""""""""""""""
+
+To start a Python Cloud Dataflow job with a Python file on GCS use
+:class:`~airflow.providers.google.cloud.operators.dataflow.DataflowCreatePythonJobOperator`.
+
+.. exampleinclude:: ../../../../airflow/providers/google/cloud/example_dags/example_dataflow.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_operator_start_python_job]
+    :end-before: [END howto_operator_start_python_job]
+
+Using a Python file on Local
+""""""""""""""""""""""""""""
+
+To start a Python Cloud Dataflow job with a Python file on local use
+:class:`~airflow.providers.google.cloud.operators.dataflow.DataflowCreatePythonJobOperator`.
+If the Python file is on GCS, you can download it to your local with
```

Comment:
You do not have to use this operator. If the file is on GCS then it can be specified using the parameter directly. We have this task at DAG, but only for testing purposes.

Reply:
[Done here](https://github.com/apache/airflow/pull/8809/commits/dc48b7a99abbe26f721bb32d1e381c3b03afb946)